hawkular-wildfly-agent/src/main/java/org/hawkular/agent/monitor/extension/MonitorServiceConfigurationBuilder.java

```diff
@@ -435,6 +436,25 @@ private static void determineMetricSetPrometheus(ModelNode config,
 
     }
 
+    private static String determinePlatformMachineId(ModelNode config, OperationContext context)
+            throws OperationFailedException {
+
+        if (!config.hasDefined(PlatformDefinition.PLATFORM)) {
+            return null; // not monitoring platform, so we don't collect machine ID
+        }
+
+        List<Property> asPropertyList = config.get(PlatformDefinition.PLATFORM).asPropertyList();
+        if (asPropertyList.size() == 0) {
+            return null; // not monitoring platform, so we don't collect machine ID
+        } else if (asPropertyList.size() > 1) {
```

Comment:
this is how wildfy builds the model node data - I believe you get an error if you don't try to read the config element here as a property list.